src/lib.rs

```diff
@@ -667,6 +667,32 @@ impl RawInvoice {
 			num_traits => None,
 		}).collect::<Vec<&Route>>()
 	}
+
+	pub fn amount_pico_btc(&self) -> Option<u64> {
+		match self.hrp.raw_amount {
+			Some(amt) => {
+				if let Some(ref mult) = self.hrp.si_prefix {
+					match mult {
+						// 10^-3
+						&SiPrefix::Milli => Some(amt * 1_000_000_000),
```

Comment:
Hardcoding the multiplier here will lead to redundancy as soon as I add functionality to automatically find the best SI prefix. I'd rather add a function `fn multiplier(&self) -> u64` to `SiPrefix`.